### ISSUE TITLE
Pass "big" binary files

### ISSUE THREAD

**Issue by col-panic** [adnanh/webhook#162]
I am trying to pass big (~25mb) binary (base64 encoded) files. While the transfer does work correctly, there seems no correct way to pass the binary string to the underlying shell script. If I use ```pass-arguments-to-command```  get ```argument list too long``` and using ```pass-environment-to-command```does not seem to change the behaviour.

Is there any way that I may store the incoming body directly into a file for subsequent processing? Or is there another way to pass binary files?

**Comment by col-panic** [comment]
I am proposing an extension like 

```
   "pass-file-to-command":
    [
      {
      	"source": "payload",
      	"name": "zippedBinary",
      	"filename": "binaryFile.zip",
      	"base64decode": true
      }
    ]
```

the rules for selection are the same like `pass-arguments-to-command` or `pass-environment-to-command` but the corresponding data is stored in a file called `filename` in location relative to `command-working-directory`. The `base64decode` option, if set to true, will transparently base64decode the respective data before storing it into `filename`.

**Comment by adnanh** [comment]
How about a different approach. Adding ability to pass/stream request value(s?) to the command's `stdin`? That way you can stream arbitrary length payload to the command/shell script.

**Comment by col-panic** [comment]
I think the streaming solution would be good for a separate implementation. On using webhook for my purpose I found that I like to transfer multiple files in one webhook call, which would not be feasible with a stream, but it is by encapsulating multiple of those as base64 data within a json object. Currently i test with files about 150mb and it works without problems. 

Using streaming one could transfer just a single stream, as AFAIK it would "eat up" all of the body containments. But for a separate patch I think it would be great!

**Comment by col-panic** [comment]
Could you please have a review on my pull request? Thanks

**Comment by adnanh** [comment]
As soon as I grab some free time. Life's been busy these days... :-(

**Comment by adnanh** [comment]
Fixed in #174
